tools/docker/static-analysis/cppcheck.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+scriptdir="$(cd "${0%/*}"; pwd)"
+rootdir="${scriptdir%/*/*/*}"
+
+. "${rootdir}/tools/docker/functions.sh"
+
+OUTPUT_FILE="$rootdir/cppcheck_results.txt"
+INCLUDES="\
+-I$rootdir/framework/platform/bpl/include \
+-I$rootdir/framework/transport/ieee1905_transport/include \
+-I$rootdir/framework/common/include \
+-I$rootdir/framework/tlvf/src/include \
+-I$rootdir/framework/tlvf/AutoGenerated/include \
+-I$rootdir/framework/discovery_agent/include \
+-I$rootdir/common/beerocks/btl/include \
+-I$rootdir/common/beerocks/tlvf/src/include \
+-I$rootdir/common/beerocks/tlvf/AutoGenerated/include \
+-I$rootdir/common/beerocks/bcl/include \
+-I$rootdir/common/beerocks/bwl/include"
+
+usage() {
+    echo "usage: $(basename "$0") <source> [source]"
+    echo "run cpp check on <source> and output number of issues per severity"
+}
+
+run_cppcheck() {
+    # --force is needed to check more than 12 ifdefs
+    cppcheck --template='{file}:{line}:{column}: {severity}: {message} [{id}]\n{code}' --force --error-exitcode=1 --enable=warning,style,information -q $INCLUDES "$@" 2>&1 | tee "$OUTPUT_FILE"
+}
+
+colorize_severity() {
+    local severity msg
+    severity="$1"
+    msg="$2"
+    case "$severity" in
+        error)
+            printf '\033[1;31m%s\033[0m\n' "$msg"
```

Comment:
Why not include `functions.sh` and use `err` `info` and friends?

Reply:
Because there is 1 color per severity type. `functions.sh` only has `err` and `success`. I didn't add them to `functions.sh` because I don't see any other use for them at the moment.